OpenRA.Mods.Common/Traits/World/Selection.cs

```diff
@@ -167,6 +167,7 @@ public bool RolloverContains(Actor a)
 		void ITick.Tick(Actor self)
 		{
 			var removed = actors.RemoveWhere(a => !a.IsInWorld || (!a.Owner.IsAlliedWith(world.RenderPlayer) && world.FogObscures(a)));
+			var cmdAlliedUnits = self.World.WorldActor.Trait<TeamTogether>().CmdAlliedUnitsEnabled;
```

Comment:
Don't do expensive trait lookups per tick or per selection. Cache it in the constructor.

Reply:
@Mailaender for this and https://github.com/OpenRA/OpenRA/pull/19144#discussion_r760478734 I need a little more guidance. Could you please point me to an example where this is already done with cached values in the constructor? This would be very much appreciated. btw many thanks for your review.

Reply:
It is done mostly everywhere. See https://github.com/OpenRA/OpenRA/blob/39250c803d9f671a0aa4d428f00c13e5ac922280/OpenRA.Mods.Common/Traits/World/Selection.cs#L40 in that same file.